compaction/compaction_manager.hh

```diff
@@ -325,6 +325,7 @@ public:
     friend class compacting_sstable_registration;
     friend class compaction_weight_registration;
     friend class compaction_manager_test;
+    friend class table;
```

Comment:
I think it's better to make the method public, than making table friend class as it will allow table to call private methods it's not supposed to.